src/screens/setup/create/CreateWalletName.js

```diff
@@ -24,34 +24,30 @@ class CreateWalletName extends Component {
   }
 
     /**
-     * Method is used to save the newly generated wallet (via ethers.js) in the global state
-     * variable and to navigate to the "generatePassphrase" screen
+     * A new wallet is created, the wallet name is passed in along with usersWallets, which will be an 
+     * empty array when user initially creates a wallet in setup.
      */
     navigate = () => {
+      const walletName = this.props.tempWalletName;
       const wallet = ethers.Wallet.createRandom();
-      this.props.newWalletCreation(wallet);
+      const userWallets = this.props.wallets;
+      this.props.initializeAppWallet(wallet, walletName, userWallets);
       const navigateToPassphrase = NavigationActions.navigate({ routeName: 'generatePassphrase' });
       this.props.navigation.dispatch(navigateToPassphrase);
     };
 
     /**
-     * Executes the action "newWalletNameEntry" with "name" as the parameter
-     * in order to update the name of the wallet in the global state variable
-     * @param {String} name
+     * The wallet name is stored in a temporary state.
      */
     getWalletName(name) {
-      this.props.newWalletNameEntry(name);
+      this.props.setTempWalletName(name);
       if (name !== '') {
         this.setState({ buttonDisabled: false });
       } else {
         this.setState({ buttonDisabled: true });
       }
```

Comment:
you can reduce this to one line. name !== "" ? this.setState({ buttonDisabled: false }) : this.setState({ buttonDisabled: false })....

My next question is if you are storing the name (presumably) in redux, why are you using state to handle a button being disabled?

Reply:
I will reducer it to one line and to the second comment I did not yet fully refactor the entire project, I was going to leave all the buttons and their disabled state to its own task. @GregTheGreek

Reply:
@GregTheGreek both issue resolved